### ISSUE TITLE
DNS are not resolved by HttpClient

### ISSUE THREAD

**Issue by jndietz** [reactor/reactor-netty#1755]
## Expected Behavior
I am expecting that `reactor-netty` will be able to resolve a DNS.

## Actual Behavior
The DNS resolution fails when validating a JWT against Okta (or calling any other downstream service) with the following exception:

```
Caused by: java.net.UnknownHostException: failed to resolve 'dev-542348.okta.com' after 3 queries 
	at io.netty.resolver.dns.DnsResolveContext.finishResolve(DnsResolveContext.java:1046) ~[netty-resolver-dns-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.resolver.dns.DnsResolveContext.tryToFinishResolve(DnsResolveContext.java:999) ~[netty-resolver-dns-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.resolver.dns.DnsResolveContext.query(DnsResolveContext.java:417) ~[netty-resolver-dns-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.resolver.dns.DnsResolveContext.access$600(DnsResolveContext.java:65) ~[netty-resolver-dns-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.resolver.dns.DnsResolveContext$2.operationComplete(DnsResolveContext.java:466) ~[netty-resolver-dns-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578) ~[netty-common-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:571) ~[netty-common-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:550) ~[netty-common-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491) ~[netty-common-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616) ~[netty-common-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:609) ~[netty-common-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:117) ~[netty-common-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.resolver.dns.DnsQueryContext.tryFailure(DnsQueryContext.java:225) ~[netty-resolver-dns-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.resolver.dns.DnsQueryContext$4.run(DnsQueryContext.java:177) ~[netty-resolver-dns-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98) ~[netty-common-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:170) ~[netty-common-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute$$$capture(AbstractEventExecutor.java:164) ~[netty-common-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java) ~[netty-common-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469) ~[netty-common-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500) ~[netty-transport-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) ~[netty-common-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.66.Final.jar:4.1.66.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.66.Final.jar:4.1.66.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]
Caused by: io.netty.resolver.dns.DnsNameResolverTimeoutException: [/205.171.3.25:53] query via UDP timed out after 5000 milliseconds (no stack trace available)
```

The IP above is a CenturyLink DNS server.

## Steps to Reproduce
I have created a sample test project that will validate a JWT against Okta.  It is a simple Spring Boot 2.5.3, Cloud 2020.0.3, Groovy 3.0, and Gradle 7.1.1. project:

https://github.com/jndietz/reactor-netty-dns-resolution

I am able to consistently reproduce this on my work PC, while on the VPN.  However, when I clone this same project and run on my personal PC, everything is fine.

https://github.com/jndietz/reactor-netty-dns-resolution

## Possible Solution
Downgrading to Spring Boot `2.3.12.RELEASE` seems to resolve the issue.

I have tried the solution provided here:

https://github.com/reactor/reactor-netty/issues/1431#issue-766946843

However, customizing `HttpClient` to use a different resolution scheme ends up requiring extra SSL configuration as well, which is not desired (and shouldn't be necessary).  This is also not desirable because I would need to customize the Okta Spring Boot starter to use different DNS resolution as well.

I believe this issue is rooted in how either the JDK or `reactor-netty` are using DNS entries from the host machine.  It also seems to only impact Windows 10.  Our applications deployed in the PCF environment are working fine.  This is impact other users in the organization using the same stack (Boot, Webflux).

## Your Environment
Windows 10
Gradle 7.1.1
Groovy 3
Spring Boot 2.5.3 (+ Webflux)
Spring Cloud 2020.0.3

* Reactor version(s) used: `reactor-netty-core` and `reactor-netty-http` version 1.0.9, `reactor-core` version 3.4.8
* Other relevant libraries versions (eg. `netty`, ...): 4.1.66 (bundled with webflux)
* JVM version (`java -version`): JDK 11 and JDK 16 (have tried both)
* OS and version (eg. `uname -a`): Windows 10

**Comment by violetagg** [comment]
@jndietz Are you able to provide some public repository?

**Comment by jndietz** [comment]
@violetagg Sure, it is public now.

I am finding that it appears to be very specific to the computer it is running on.  I’m not sure it will show the error on your machine.

I am going to try and do some tests with my company’s VPN today to see if that may be causing an issue.

**Comment by violetagg** [comment]
@jndietz one additional question: you do not switch to VPN while the app is working don't you?

**Comment by jndietz** [comment]
@violetagg 

I am noticing that this has to be related to the VPN my company uses.  I really can't explain what is happening, but I noticed while I was on the VPN, the calls to Okta failed.  When I disconnected from the VPN, the calls would succeed.  I did notice that using Spring Boot 2.3.x would resolve the issue while on the VPN.

I'm going to close this issue, because I believe it is heavily dependent on the environment.  Though something did change between netty 0.9 and 1.0 that is causing intermittent issues as well.
